### PR TITLE
validator - check asset.gltfUpAxis

### LINKED ISSUE #92: Validator - check asset.gltfUpAxis
@rms13

A tileset may declare the up-axis used by glTF models embedded in its tiles, for example:

```
{
    asset: {
        version: '0.0',
        gltfUpAxis: 'Z'
    }
}
```

If `gltfUpAxis` is defined its value should either be `'X'`, `'Y'`, or `'Z'`. Anything else should return an error message.

The code changes will go in `validateTileset.js`. Also add a test to `validateTilesetSpec`. You can model the test after `returns error message when the top-level geometricError is missing`.

### PR DESCRIPTION
**Pull request by rms13**
Fixes #92. 

First checks if gltfUpAxis is defined. If it is, checks if it is 'X', 'Y', or 'Z'. Changes are in validateTileset.js
Added a test in validateTilesetSpec for the check.

### CODE REVIEW

File: validator/lib/validateTileset.js [hunk 1]
```diff
@@ -35,6 +35,12 @@ function validateTopLevel(tileset) {
     if (!defined(tileset.geometricError)) {
         return 'Tileset must declare its geometricError as a top-level property.';
     }
+
+    if (defined(tileset.asset.gltfUpAxis)) {
+        if (tileset.asset.gltfUpAxis !== 'X' && tileset.asset.gltfUpAxis !== 'Y' && tileset.asset.gltfUpAxis !== 'Z') {
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Create a variable for `tileset.asset.gltfUpAxis` so it doesn't have to be repeated multiple times.

File: validator/lib/validateTileset.js [hunk 2]
```diff
@@ -35,6 +35,12 @@ function validateTopLevel(tileset) {
...
+            return 'gltfUpAxis declared under asset should either be X, Y, or Z.';
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Tweak the message a bit: `'gltfUpAxis should either be "X", "Y", or "Z".'`

**Comment by rms13** [comment]
@lilleyse 
Fixed. Thanks for the corrections!

**Comment by lilleyse** [comment]
Your changes look good, I just merged https://github.com/AnalyticalGraphicsInc/3d-tiles-tools/pull/93 so you'll need to resolve the merge conflicts.

**Comment by lilleyse** [comment]
In the future it is best to work off of a branch of `validator` instead of working directly on the `validator` branch. This will make it easier to fix merge conflicts.

File: validator/lib/validateTileset.js [hunk 3]
```diff
@@ -36,6 +36,13 @@ function validateTopLevel(tileset) {
         return 'Tileset must declare its geometricError as a top-level property.';
     }
 
+    var gltfUpAxis = tileset.asset.gltfUpAxis;
+    if (defined(gltfUpAxis)) {
+        if (gltfUpAxis !== 'X' && gltfUpAxis !== 'Y' && gltfUpAxis !== 'Z') {
+            return 'gltfUpAxis should either be "X", "Y", or "Z".';
+        }
+    }
+
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Probably best to put this code after the asset checks since it requires `tileset.asset` to exist.

Comment (rms13) [inline validator/lib/validateTileset.js]: Makes sense. Changed the order here and also in tests (to maintain consistency).

**Comment by rms13** [comment]
Had conflicts between branches on my fork, so merged them and resolved the conflicts.
Please review the changes before merging with the original repo.
Fixes #96 and #94

File: validator/lib/validateTileset.js [hunk 4]
```diff
@@ -28,7 +28,12 @@ function validateTileset(tileset, tilesetDirectory) {
         return Promise.resolve(message);
     }
 
-    return validateTileHierarchy(tileset.root, tilesetDirectory);
+    message = validateTileHierarchy(tileset.root, tilesetDirectory);
+    if (defined(message)) {
+        return Promise.resolve(message);
+    }
+
+    return undefined;
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: `Promise.resolve(validateTileHierarchy(tileset.root, tilesetDirectory));` is equivalent here.

Comment (rms13) [inline validator/lib/validateTileset.js]: Corrected!

**Comment by lilleyse** [comment]
Thanks, all looks good.

**Pull request merged**